core/common/src/main/java/alluxio/PropertyKey.java

```diff
@@ -795,6 +796,7 @@
         "alluxio.user.ufs.block.read.concurrency.max";
     public static final String USER_UFS_BLOCK_OPEN_TIMEOUT_MS =
         "alluxio.user.ufs.block.open.timeout.ms";
+    public static final String USER_SHORT_CIRCUIT_ENABLED = "alluxio.user.short.circuit.disabled";
```

Comment:
I think this can be a useful site property for both test and non-test code. 
1. For test code, our integration tests do not seem to have a good coverage for remote read/writes.  For example, RemoteBlock{In,Out}StreamIntegrationTest, File{In,Out}StreamIntegrationTest, neither of them do remote operations. This site property makes it easier to test this.

2. It can be useful for non-test code if people do not want to have short-circuit read for any reasons we have known. 

One way to get rid of this flag is to expose somehow override the localhost with WhiteBox. It is not as clean as this but i am fine with that if you prefer.